datasets/samplemotordatalimerick.py

```diff
@@ -0,0 +1,662 @@
+
+        self.sensor_selected = sensor_selected
+        self.rpm_selected = rpm_selected
+
+        self.num_of_features = 3
+
+        processed_folder = \
+            os.path.join(root, self.__class__.__name__, 'processed')
+
+        self.__makedir_exist_ok(processed_folder)
+
+        self.specs_identifier = f'eval_mode_{self.eval_mode}_' + \
+                                f'label_as_signal_{self.label_as_signal}_' + \
+                                f'ds_{self.downsampling_ratio}_' + \
+                                f'dur_{self.signal_duration_in_sec}_' + \
+                                f'ovlp_ratio_{self.overlap_ratio}_' + \
+                                f'random_split_{self.random_or_speed_split}_' + \
+                                f'sensor_selected_{self.sensor_selected}_' +\
+                                f'rpm_{self.rpm_selected}'
+
+        train_dataset_pkl_file_path = \
+            os.path.join(processed_folder, f'train_{self.specs_identifier}.pkl')
+
+        test_dataset_pkl_file_path =  \
+            os.path.join(processed_folder, f'test_{self.specs_identifier}.pkl')
+
+        if self.d_type == 'train':
+            self.dataset_pkl_file_path = train_dataset_pkl_file_path
+
+        elif self.d_type == 'test':
+            self.dataset_pkl_file_path = test_dataset_pkl_file_path
+
+        self.signal_list = []
+        self.lbl_list = []
+
+        self.__create_pkl_files()
+        self.is_truncated = False
+
+    def __create_pkl_files(self):
+        if os.path.exists(self.dataset_pkl_file_path):
+
+            print('\nPickle files are already generated ...\n')
+
+            (self.signal_list, self.lbl_list) = \
+                pickle.load(open(self.dataset_pkl_file_path, 'rb'))
+            return
+
+        self.__gen_datasets()
+
+    def normalize_signal(self, features):
+        """
+        Normalize signal with Local Min Max Normalization
+        """
+        # Normalize data:
+        for instance in range(features.shape[0]):
+            instance_max = np.max(features[instance, :, :], axis=1)
+            instance_min = np.min(features[instance, :, :], axis=1)
+
+            for feature in range(features.shape[1]):
+                for signal in range(features.shape[2]):
+                    features[instance, feature, signal] = (
+                        (features[instance, feature, signal] - instance_min[feature]) /
+                        (instance_max[feature] - instance_min[feature])
+                    )
+
+        return features
+
+    def __gen_datasets(self):
+        print('\nGenerating dataset pickle files from the raw data '
+              f'files (specs identifier: {self.specs_identifier}) ...\n')
+
+        actual_root_dir = os.path.join(self.root, self.__class__.__name__,
+                                       "SpectraQuest_Rig_Data_Voyager_3/")
+
+        data_dir = os.path.join(actual_root_dir, f'Test_Results_Data_{self.sensor_selected}/')
```

Comment:
Most probably the csv files folder will not have Test_Results_ prefix

Reply:
Also a check here wrt data_dir would be nice to add, can also be added after dataset repo is published. If folder does not exist or empty data loader will return 0 samples wo any control

Reply:
Updated